### PR TITLE
Modify the signature and the way the derivationHash is saved.

### PR DESCRIPTION
**Pull request by pamgonzalez**
Previously only the derivationHash was saved, now the hash of the original bitcoin transaction is added as a joint key.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 1]
```diff
@@ -600,26 +601,36 @@ private void saveCoinbaseInformations() {
...
+        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationArguments(btcTxHash, derivationArgsHash));
 
         return ((data != null) && (data.length == 1) && (data[0] == FAST_BRIDGE_FEDERATION_SCRIPT_HASH_TRUE_VALUE));
     }
 
-    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash derivationArgsHash) {
+    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash btcTxHashToSave, Sha256Hash derivationArgsHash) {
         if (activations.isActive(RSKIP176)) {
+            fastBridgeBtcTxHashToSave = btcTxHashToSave;
             fastBridgeDerivationArgumentsScriptHashToSave = derivationArgsHash;
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:621]: Why `fastBridgeDerivationArgumentsScriptHashToSave`? Should not be `fastBridgeDerivationArgumentsHashToSave`?

**Review by guidohernan93 — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java [hunk 2]
```diff
@@ -2154,13 +2154,16 @@ public void saveCoinBaseInformation_after_RSKIP143() throws IOException {
     public void isFastBridgeFederationDerivationHashUsed_afterRSKIP176_returnTrue() {
         Repository repository = mock(Repository.class);
 
-        Sha256Hash hash = Sha256Hash.ZERO_HASH;
+        Sha256Hash derivationHash = PegTestUtils.createHash(1);
+        Sha256Hash rskTxHash = PegTestUtils.createHash(2);
```
Comment (josedahlquist) [inline rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java:2158]:
```suggestion
        Sha256Hash btcTxHash = PegTestUtils.createHash(2);
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java [hunk 3]
```diff
@@ -2281,24 +2297,72 @@ public void saveDerivationArgumentsScriptHash_afterRSKIP176_Ok() throws IOExcept
...
+
+        provider.markFastBridgeFederationDerivationHashAsUsed(rskTxHash, derivationHash);
+
+        provider.save();
+        
+        verifyZeroInteractions(repository);
+    }
+
+    @Test
+    public void saveDerivationArgumentsScriptHash_afterRSKIP176_nullrskTxHash_notSaved() throws IOException {
```
Comment (josedahlquist) [inline rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java:2337]:
```suggestion
    public void saveDerivationArgumentsScriptHash_afterRSKIP176_null_derivationHash_notSaved() throws IOException {
```

**Review by josedahlquist — APPROVED** [review]
There are some typos in the tests, other than that LGTM

File: rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java [hunk 4]
```diff
@@ -2281,24 +2297,48 @@ public void saveDerivationArgumentsScriptHash_afterRSKIP176_Ok() throws IOExcept
...
                 PrecompiledContracts.BRIDGE_ADDR,
-                DataWord.fromLongString("fastBridgeScriptHash-" + hash.toString()),
+                DataWord.fromLongString("fastBridgeHashUsedInBtcTx-" + btcTxHash.toString() + derivationHash.toString()),
                 new byte[]{FAST_BRIDGE_FEDERATION_SCRIPT_HASH_TRUE_VALUE_TEST}
         );
         verifyNoMoreInteractions(repository);
     }
 
+    @Test
+    public void saveDerivationArgumentsScriptHash_afterRSKIP176_nullBtcTxHash_notSaved() throws IOException {
```
Comment (josedahlquist) [inline rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java:2313]:
```suggestion
    public void saveDerivationArgumentsScriptHash_afterRSKIP176_null_derivationHash_notSaved() throws IOException {
```

**Review by josedahlquist — APPROVED** [review]

**Pull request merged**